spec/models/miq_report_result_spec.rb

```diff
@@ -1,14 +1,56 @@
 describe MiqReportResult do
-  context "persisting generated report results" do
-    before(:each) do
-      EvmSpecHelper.local_miq_server
+  before :each do
+    EvmSpecHelper.local_miq_server
+
+    @user1 = FactoryGirl.create(:user_with_group)
+  end
+
+  context "report result created by User 2 with current group 2" do
+    before :each do
+      @report = FactoryGirl.create(:miq_report)
+      group_2 = FactoryGirl.create(:miq_group)
+      @user1.miq_groups << group_2
+      @report_result1 = FactoryGirl.create(:miq_report_result, :miq_report_id => @report.id, :miq_group => group_2)
+      @report_result2 = FactoryGirl.create(:miq_report_result, :miq_report_id => 999_999_999)
+      User.current_user = @user1
+    end
+
+    describe ".with_report" do
+      it "returns only requested report results" do
+        report_result = MiqReportResult.with_report(@report_result1.miq_report_id)
+        expect(report_result).to include(@report_result1)
+      end
+
+      it "returns all requested report results" do
+        report_result = MiqReportResult.with_report
+        expect(report_result).to include(@report_result1)
+        expect(report_result).to include(@report_result2)
+      end
+    end
 
-      @user = FactoryGirl.create(:user_with_group)
+    describe ".with_current_user_groups" do
+      it "returns report results by generated by user 2, non-admin user logged" do
+        report_result = MiqReportResult.with_current_user_groups
+        expect(report_result).to include(@report_result1)
+        expect(report_result).not_to include(@report_result2)
+      end
 
+      it "returns report all results, admin user logged" do
+        admin_role = FactoryGirl.create(:miq_user_role, :name => "EvmRole-administrator", :read_only => false)
+        User.current_user.current_group.miq_user_role = admin_role
+        report_result = MiqReportResult.with_current_user_groups
+        expect(report_result).to include(@report_result1)
+        expect(report_result).to include(@report_result2)
+      end
+    end
+  end
+
+  context "persisting generated report results" do
+    before(:each) do
       5.times do |i|
         vm = FactoryGirl.build(:vm_vmware)
-        vm.evm_owner_id = @user.id               if i > 2
-        vm.miq_group_id = @user.current_group.id if vm.evm_owner_id || (i > 1)
+        vm.evm_owner_id = @user1.id               if i > 2
+        vm.miq_group_id = @user1.current_group.id if vm.evm_owner_id || (i > 1)
```

Comment:
We should see if we can remove this 5.times setup of all of these vms.  It feels like it's more setup than we really need for the tests.  This can be done in a followup PR.